python/Ganga/GPIDev/Credentials2/VomsProxy.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import absolute_import
+
+import os
+import re
+from datetime import datetime, timedelta
+
+import Ganga.Utility.logging
+from Ganga.GPIDev.Schema import SimpleItem
+from Ganga.Utility.Shell import Shell
+from Ganga.Utility.Config import getConfig
+
+from .ICredentialInfo import ICredentialInfo, cache
+from .ICredentialRequirement import ICredentialRequirement
+from .exceptions import CredentialRenewalError
+
+logger = Ganga.Utility.logging.getLogger()
+
+
+class VomsProxyInfo(ICredentialInfo):
+    """
+    A wrapper around a voms proxy file
+    """
+
+    def __init__(self, requirements, check_file=False, create=False):
+        self._shell = None
+
+        super(VomsProxyInfo, self).__init__(requirements, check_file, create)
+
+    def create(self):
+        """
+        Creates the grid proxy.
+
+        Raises:
+            CredentialRenewalError: If the renewal process returns a non-zero value
+        """
+        voms_command = ''
+        logger.debug('require ' + self.initial_requirements.vo)
+        if self.initial_requirements.vo:
+            voms_command = '-voms %s' % self.initial_requirements.vo
+            if self.initial_requirements.group or self.initial_requirements.role:
+                voms_command += ':'
+                if self.initial_requirements.group:
+                    voms_command += '/%s' % self.initial_requirements.group
+                if self.initial_requirements.role:
+                    voms_command += '/%s' % self.initial_requirements.role
+        logger.debug(voms_command)
+        command = 'voms-proxy-init -pwstdin -out %s %s' % (self.location, voms_command)
+        logger.debug(command)
+        return_code = self.shell.system(command)
+        if return_code == 0:
+            logger.debug('Grid proxy %s created. Valid for %s', self.location, self.time_left())
+        else:
+            raise CredentialRenewalError('Failed to create VOMS proxy')
+
+    @property
+    def shell(self):
+        if self._shell is None:
```

Comment:
Because I want to be explicit about a missing value. I want to explicitly check for an unassigned value rather than checking for truthiness.